src/libraries/Common/src/System/Text/DBCSDecoder.cs

```diff
@@ -201,17 +200,17 @@ public override unsafe int GetChars(byte* bytes, int byteCount, char* chars, int
             ArgumentNullException.ThrowIfNull(bytes);
             ArgumentNullException.ThrowIfNull(chars);
 
-            if (byteIndex < 0 || byteCount < 0)
-                throw new ArgumentOutOfRangeException(byteIndex < 0 ? nameof(byteIndex) : nameof(byteCount), SR.ArgumentOutOfRange_NeedNonNegNum);
+            ArgumentOutOfRangeException.ThrowIfNegative(byteIndex);
 
-            if (charIndex < 0 || charCount < 0)
-                throw new ArgumentOutOfRangeException(charIndex < 0 ? nameof(charIndex) : nameof(charCount), SR.ArgumentOutOfRange_NeedNonNegNum);
+            ArgumentOutOfRangeException.ThrowIfNegative(byteCount);
 
-            if (bytes.Length - byteIndex < byteCount)
-                throw new ArgumentOutOfRangeException(nameof(bytes), SR.ArgumentOutOfRange_IndexCountBuffer);
+            ArgumentOutOfRangeException.ThrowIfNegative(charIndex);
 
-            if (chars.Length - charIndex < charCount)
-                throw new ArgumentOutOfRangeException(nameof(chars), SR.ArgumentOutOfRange_IndexCountBuffer);
+            ArgumentOutOfRangeException.ThrowIfNegative(charCount);
+
+            ArgumentOutOfRangeException.ThrowIfLessThan(bytes.Length - byteIndex, byteCount);
+
+            ArgumentOutOfRangeException.ThrowIfLessThan(chars.Length - charIndex, charCount);
```

Comment:
Some relatively simple checks here turned into a lot of method calls - maybe inlined - is the generated code larger?

Reply:
I didn't check that, what's the easiest way to do that?
Also current implementation should use same private helper as `ArgumentNullException` for better inlining I guess
https://github.com/dotnet/runtime/blob/c9f80fdde29c71ac87971383c77a1a5e665fadb9/src/libraries/System.Private.CoreLib/src/System/ArgumentNullException.cs#L90-L92